search-commons/src/main/java/no/unit/nva/search2/ResourceQuery.java

```diff
@@ -0,0 +1,133 @@
+package no.unit.nva.search2;
+
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static no.unit.nva.search2.ResourceParameter.OFFSET;
+import static no.unit.nva.search2.ResourceParameter.PAGE;
+import static no.unit.nva.search2.ResourceParameter.PER_PAGE;
+import static no.unit.nva.search2.ResourceParameter.keyFromString;
+import static no.unit.nva.search2.constant.Defaults.DEFAULT_VALUE_PAGE;
+import static no.unit.nva.search2.constant.Defaults.DEFAULT_VALUE_PER_PAGE;
+import static no.unit.nva.search2.constant.Defaults.DEFAULT_VALUE_SORT;
+import static no.unit.nva.search2.constant.Defaults.DEFAULT_VALUE_SORT_ORDER;
+
+import no.unit.nva.search2.common.OpenSearchQuery;
+import no.unit.nva.search2.model.PagedSearchResponseDto;
+import no.unit.nva.search2.common.QueryBuilder;
+import no.unit.nva.search2.model.OpenSearchSwsResponse;
+
+import java.util.stream.Stream;
+
+import nva.commons.apigateway.exceptions.BadGatewayException;
+import org.jetbrains.annotations.NotNull;
+
+public final class ResourceQuery extends OpenSearchQuery<ResourceParameter,PagedSearchResponseDto> {
+
+    @Override
+    public PagedSearchResponseDto doSearch(@NotNull SwsOpenSearchClient queryClient) throws BadGatewayException {
+
+        return
+            Stream.of(queryClient.doSearch(openSearchUri()))
+                .map(this::toPagedSearchResponseDto)
+                .findFirst().orElseThrow();
+    }
+
+    private ResourceQuery() {
+        super();
+    }
+
+    @SuppressWarnings("PMD.NullAssignment")
+    @NotNull
+    private PagedSearchResponseDto toPagedSearchResponseDto(@NotNull OpenSearchSwsResponse response) {
```

Comment:
this seems to transform data between internal and client model, but is this the one being used in handler?

Reply:
yes

Reply:
```
    public PagedSearchResponseDto doSearch(@NotNull SwsOpenSearchClient queryClient) throws BadGatewayException {
        return
            Stream.of(queryClient.doSearch(openSearchUri()))
                .map(this::toPagedSearchResponseDto)
                .findFirst().orElseThrow();
    }
```
10 lines above.

Reply:
im just confused because of the overrides